src/util/schnorr.rs

```diff
@@ -17,3 +17,64 @@
 //!
 
 pub use secp256k1::schnorrsig::{PublicKey, KeyPair};
+use secp256k1::{Secp256k1, Verification};
+use hashes::{Hash, HashEngine};
+use util::taproot::{TapBranchHash, TapTweakHash};
+
+/// Untweaked Schnorr public key
+pub struct UntweakedPublicKey(pub PublicKey);
```

Comment:
The tuple struct's fiels is set to public to help with initialization at `src/util/address.rs line 1190`.

There could use a method instead of directly initializing though.

Reply:
I also don't mind having 

pub type UnTweakedPublicKey = schnorr::PublicKey;

I think in all cases we should deal with untweaked keys and let the library handle all tweaking.